### PR TITLE
Update VACOLS location when ChangeHearingRequestTypeTask is completed

### PR DESCRIPTION
**Pull request by rubaiyat22**
Connects #15427

### Description
- update location of legacy appeals to caseflow at completion of task instead of when the task is initially created
- update test

### Code Documentation Updates
- [x] Add or update code comments at the top of the class, module, and/or component.

### CODE REVIEW

File: app/models/tasks/change_hearing_request_type_task.rb [hunk 1]
```diff
@@ -70,6 +73,11 @@ def update_appeal_and_self(payload_values, params)
     multi_transaction do
       appeal.update!(changed_request_type: payload_values[:changed_request_type])
 
+      # update location to `CASEFLOW`
+      if (appeal.is_a?(LegacyAppeal))
```
Comment (rubaiyat22) [inline app/models/tasks/change_hearing_request_type_task.rb:77]: added this check for future compatibality with ama appeal but i'm not sure it's necessary. open to feedback if should keep or remove

File: app/models/tasks/change_hearing_request_type_task.rb [hunk 2]
```diff
@@ -70,6 +73,11 @@ def update_appeal_and_self(payload_values, params)
     multi_transaction do
       appeal.update!(changed_request_type: payload_values[:changed_request_type])
 
+      # update location to `CASEFLOW`
+      if appeal.is_a?(LegacyAppeal)
```
Comment (rubaiyat22) [inline app/models/tasks/change_hearing_request_type_task.rb:77]: added this check for future compatibality with ama appeal but i'm not sure it's necessary. open to feedback if should keep or remove

**Review by tomas-nava — APPROVED** [review]
approve!

File: app/models/tasks/change_hearing_request_type_task.rb (hunk 2 above)
Comment (tomas-nava) [reply app/models/tasks/change_hearing_request_type_task.rb:77]: I'm fine keeping it in, better to be overly cautious.

**Pull request merged**